pkg/controller/operators/olm/requirements.go

```diff
@@ -408,15 +407,15 @@ func (a *Operator) isGVKRegistered(group, version, kind string) error {
 	resources, err := a.opClient.KubernetesInterface().Discovery().ServerResourcesForGroupVersion(gv.String())
 	if err != nil {
 		logger.WithField("err", err).Info("could not query for GVK in api discovery")
-		return err
+		return false, err
 	}
 
 	for _, r := range resources.APIResources {
 		if r.Kind == kind {
-			return nil
+			return true, nil
 		}
 	}
 
 	logger.Info("couldn't find GVK in api discovery")
-	return olmErrors.GroupVersionKindNotFoundError{group, version, kind}
+	return false, nil
```

Comment:
I could not find any usage of `isGVKRegistered` where the actual content of the error was used, so I changed the function signature to separate the found/not found question from any errors that may have occurred trying to arrive at the answer.

Any caller of `isGVKRegistered` could trivially re-construct the previous behavior with:
```go
func (a *Operator) oldIsGVKRegistered(group, version, kind string) error {
	ok, err := a.isGVKRegistered(group, version, kind)
	if err != nil {
		return err
	}
	if !ok {
		return olmErrors.GroupVersionKindNotFoundError{group, version, kind}
	}
	return nil
}